pkg/sources/adapter/azureeventhubsource/adapter.go

```diff
@@ -92,25 +79,12 @@ func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClie
 
 	env := envAcc.(*envConfig)
 
-	var hub *eventhub.Hub
-	var authErr error
-
-	switch {
-	case env.ConnStr != "":
-		hub, authErr = newHubWithConnStr(env)
-	case env.HubKeyName != "":
-		hub, authErr = newHubWithKey(env)
-	case env.TenantID != "":
-		hub, authErr = newHubWithSP(env)
-	default:
-		logger.Panic("neither SAS nor JWT auth env variables were detected")
-	}
-
-	if authErr != nil {
-		logger.Panic(authErr)
+	hub, err := eventhub.NewHubFromEnvironment()
+	if err != nil {
+		logger.Panicw("Unable to create Event Hub client", zap.Error(err))
 	}
```

Comment:
As a nice little bonus (I think), the Azure SDK will figure out the Event Hub, as well as the auth scheme, based solely on what the controller exported in the environment. No more complicated logic.